### PR TITLE
UCT/IB: Compact AV fix

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5.c [hunk 1]
```diff
@@ -91,6 +91,7 @@ ucs_status_t uct_ib_mlx5_get_compact_av(uct_ib_iface_t *iface, int *compact_av)
     }
 
     uct_ib_iface_fill_ah_attr_from_addr(iface, ib_addr, iface->path_bits[0], &ah_attr);
+    ah_attr.is_global = 0;
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.c]: pls add comment why it's done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8087/ for details (Mellanox internal link).

**Comment by brminich** [comment]
Errors look relevant:
```
17:24:35 #   Device: mlx5_1:1
17:24:35 [hpc-test-node2:3475 :0:3475]    ib_iface.c:286  ERROR: ibv_create_ah(dlid=0 sl=0 port=1 src_path_bits=0) on mlx5_1:1 failed: Invalid argument
17:24:35 
17:24:35 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/uct/ib/base/ib_iface.c: [ uct_ib_iface_create_ah() ]
17:24:35       ...
17:24:35       282         }
17:24:35       283 
17:24:35       284         ucs_error("ibv_create_ah(%s) on "UCT_IB_IFACE_FMT" failed: %m", buf,
17:24:35 ==>   285                   UCT_IB_IFACE_ARG(iface));
17:24:35       286         return UCS_ERR_INVALID_ADDR;
17:24:35       287     }
17:24:35       288 
17:24:35 
17:24:35 ==== backtrace ====
17:24:35  0 0x000000000002981d uct_ib_iface_create_ah()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/uct/ib/base/ib_iface.c:285
17:24:35  1 0x0000000000031fdb uct_ib_mlx5_get_compact_av()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/uct/ib/mlx5/ib_mlx5.c:95
17:24:35  2 0x00000000000a6d46 uct_ud_mlx5_iface_t_init()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/uct/ib/ud/accel/ud_mlx5.c:721
17:24:35  3 0x000000000001fdc6 uct_iface_open()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/uct/base/uct_md.c:352
17:24:35  4 0x0000000000402f81 print_iface_info()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/tools/info/tl_info.c:148
17:24:35  5 0x000000000040488a print_tl_info()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/tools/info/tl_info.c:357
17:24:35  6 0x0000000000404b44 print_uct_info()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/tools/info/tl_info.c:491
17:24:35  7 0x0000000000402034 main()  /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/1/contrib/../src/tools/info/ucx_info.c:182
17:24:35  8 0x0000000000021c05 __libc_start_main()  ???:0
17:24:35  9 0x0000000000402116 _start()  ???:0
17:24:35 ===================
```

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8114/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8115/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8116/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8117/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8133/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8135/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8149/ for details (Mellanox internal link).

**Comment by Artemy-Mellanox** [comment]
@yosefe

File: src/uct/ib/base/ib_iface.h [hunk 2]
```diff
@@ -492,8 +492,9 @@ void uct_ib_iface_fill_ah_attr_from_gid_lid(uct_ib_iface_t *iface, uint16_t lid,
     ah_attr->port_num          = iface->config.port_num;
     ah_attr->grh.traffic_class = iface->config.traffic_class;
 
-    if (iface->is_global ||
-        iface->gid.global.subnet_prefix != gid->global.subnet_prefix) {
+    if (iface->is_global_addr ||
+        (iface->gid.global.subnet_prefix != gid->global.subnet_prefix)) {
+        ucs_assert(gid->global.interface_id != 0);
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.h]: maybe assert_always?

File: src/uct/ib/cm/cm_ep.c [hunk 3]
```diff
@@ -54,8 +54,13 @@ static ucs_status_t uct_cm_ep_fill_path_rec(uct_cm_ep_t *ep,
     path->sgid                      = iface->super.gid;
     path->dlid                      = htons(ep->dlid);
     path->slid                      = htons(uct_ib_iface_port_attr(&iface->super)->lid);
-    path->dgid                      = ep->dgid;
-    path->hop_limit                 = iface->super.config.hop_limit;
+    if (iface->super.is_global_addr) {
+        path->dgid                  = ep->dgid;
```
Comment (yosefe) [inline src/uct/ib/cm/cm_ep.c]: ucs_assert_always(gid->global.interface_id != 0);

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8156/ for details (Mellanox internal link).

**Comment by brminich** [comment]
#3009 

bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8165/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8168/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8172/ for details (Mellanox internal link).

**Pull request merged**